### ISSUE TITLE
.RPL Files: paths are not relative?

### ISSUE THREAD

**Issue by LexaProductions** [reaper-oss/sws#929]
Hi
Is there anything that can be done to make the Reaper Project list relative to a master Folder? I would like to use the same Project list on more than one computer and it's a challenge to always go into the RPL files to replace the proper Paths.

Thanks For you time.

**Comment by mdmayfield** [comment]
I would also really appreciate this functionality.

It looks like it would need to be implemented in https://github.com/reaper-oss/sws/blob/master/Projects/ProjectList.cpp, but I don't know ReaScript well enough to say the best method.

It looks like EnumProjects returns a full path for the filename. I don't know if [GetProjectPath](https://www.cockos.com/reaper/sdk/reascript/reascripthelp.html#GetProjectPath) returns a full path+filename or just the enclosing folder. Either way though I cringe at the thought of kludging in some kind of text manipulation to do this.

**Comment by LexaProductions** [comment]
I'd like to bump this. Anybody?

**Comment by cfillion** [comment]
Project list paths can already be relative to the REAPER process' working directory (by editing the .RPL in a text editor). On Windows the working directory is set to the path in the `lastcwd` setting in REAPER.ini. On mac it is / (root) unless launched from terminal.

Not sure if this helps though. I suppose the idea here is to have project list optionally load and save projects relative to the RPL file's path?

**Comment by LexaProductions** [comment]
Thank you very much CFillion

I'm not sure I understand the answer though. Maybe I did not ask the right question:
My goal is to:
Put an RPL file in a master folder containing several Reaper projects. 
/Volume/ComputerA/.../MasterFolder/RPLFile.RPL
This File would point to projects that are located in:
/Volume/ComputerA/.../MasterFolder/
Then be able to copy this master folder to any computer and have Reaper being able to open this RPL file 
/Volume/ComputerB/.../MasterFolder/RPLFile.RPL
As long as the projects are located in:
/Volume/ComputerB/.../MasterFolder

I hope my question makes more sense?

**Comment by cfillion** [comment]
![Demo](https://i.imgur.com/OEDgbhs.gif)

**Comment by LexaProductions** [comment]
Ok I guess this is an advance but not sure it is quite what I need?
If I have several projects that are all in separate folders (All Those folders being a child from the Master Folder) How does this works.  I don't quite understand what the CD command does in the Terminal.  I thought that was a simple "Change Directory" command?

**Comment by cfillion** [comment]
Yes `cd` is change directory. It changes the directory from which programs (here it's REAPER) are run next. If you set the directory to /Volumes/ComputerX/blabla/MasterFolder and have a line "ChildFolder/For/Project.RPP" in the .RPL SWS will attempt to load /Volumes/ComputerX/blabla/MasterFolder/ChildFolder/For/Project.RPP.

(This is pretty much a temporary solution short of rewriting that SWS feature to allow relative loads the proper way.)

**Comment by LexaProductions** [comment]
Hello CFillion.  Any hope about this feature to be implementing properly?

**Comment by LexaProductions** [comment]
I believe THIS is what needs to be modified:
https://github.com/reaper-oss/sws/blob/master/Projects/ProjectMgr.cpp

**Comment by smadep** [comment]
+1 from me;
relative paths would be great, but I also understand that absolute paths are less error-prone, as open projects could sit in totally different directories.

in my use case the RPL file sits in a different folder from the projects, so I would even need **..\..\**xyz\project1.rpp "logic".
that would increase complexity to check which part of the paths is shared across all projects which should go into the list. and only if there is a common part, make the paths relative to the (saving location of) RPL file :-)

will also add two more related "issues"/feature requests about 

- associating RPL extension with reaper => open RPL right from browser with reaper ... don't know whether this is even possible
- "open project list" using last folder, like "open project" and not always some user-profile-folder
